app/controllers/houseConsignment/index/items/CombinedNomenclatureCodeController.scala

```diff
@@ -0,0 +1,74 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.houseConsignment.index.items
+
+import controllers.actions._
+import forms.CombinedNomenclatureCodeFormProvider
+import models.requests.DataRequest
+import models.{ArrivalId, Index, Mode}
+import pages.houseConsignment.index.items.CombinedNomenclatureCodePage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.houseConsignment.index.items.CombinedNomenclatureCodeView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CombinedNomenclatureCodeController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  actions: Actions,
+  formProvider: CombinedNomenclatureCodeFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: CombinedNomenclatureCodeView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId) {
+      implicit request =>
+        val preparedForm = request.userAnswers.get(CombinedNomenclatureCodePage(houseConsignmentIndex, itemIndex)) match {
+          case None        => formProvider(houseConsignmentIndex, itemIndex)
+          case Some(value) => formProvider(houseConsignmentIndex, itemIndex).fill(value)
+        }
+
+        Ok(view(preparedForm, request.userAnswers.mrn, arrivalId, houseConsignmentIndex, itemIndex, isXI, mode))
+    }
+
+  def onSubmit(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId).async {
+      implicit request =>
+        formProvider(houseConsignmentIndex, itemIndex)
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              Future.successful(BadRequest(view(formWithErrors, request.userAnswers.mrn, arrivalId, houseConsignmentIndex, itemIndex, isXI, mode))),
+            value =>
+              for {
+                updatedAnswers <- Future.fromTry(request.userAnswers.set(CombinedNomenclatureCodePage(houseConsignmentIndex, itemIndex), value))
+                _              <- sessionRepository.set(updatedAnswers)
+              } yield Redirect(controllers.routes.UnloadingFindingsController.onPageLoad(arrivalId))
+          )
+    }
+
+  private def isXI(implicit request: DataRequest[_]): Boolean =
+    request.userAnswers.ie043Data.CustomsOfficeOfDestinationActual.referenceNumber.startsWith("XI")
```

Comment:
This looks familiar 😄 Could we maybe add a method to the implicit class at the bottom of `models.package`? Something like `def hasXIOfficeOfDestination: Boolean = value.CustomsOfficeOfDestinationActual.referenceNumber.startsWith("XI")` then you can just use `request.userAnswers.ie043Data.hasXIOfficeOfDestination` in the controllers